goss-testing/scripts/iscsi_sbps_sanity.sh

```diff
@@ -0,0 +1,131 @@
+# projection.
+
+# shellcheck disable=SC2207,SC2143,SC2002
+
+LIO_ACT=0
+
+echo "Checking whether the node is configured with iSCSI"
+
+ISCSI=$(kubectl get nodes --selector="iscsi=sbps,kubernetes.io/hostname=${HOSTNAME}" -o jsonpath='{.items[*].metadata.name}')
+
+if [ -z "${ISCSI}" ]
+then
+    echo "iSCSI is not configured, exiting"
+    exit 0
+else
+    echo "${HOSTNAME} is configured with iSCSI"
+fi
+
+# Verifying whether Marshal agent is running or not
+
+echo "*******************************"
+echo "Marshal agent is $(systemctl is-active sbps-marshal.service)"
+echo "*******************************"
+
+# Verifying whether LIO target(s) is active
+
+LIO_ACT=$(targetcli /iscsi status | awk '{print $(NF)}')
+
+if [[ "${LIO_ACT}" -gt 0 ]]
+then
+    echo "${LIO_ACT} LIO Target(s) is active"
+else
+    echo "LIO Target(s) is not active"
+fi
+
+echo "*******************************"
+
+# Verify TCP service probes complete against all active portals (iSCSI)
+
+PORTALS=($(ss -tnpl | grep ':3260 ' | awk '{ print $4 }' ))
+
+if [[ -n "${PORTALS[*]}" ]]; then
+    echo "iSCSI portals exist and are as below:"
+    echo "-------------------------------"
+    for portals in "${PORTALS[@]}"
+    do
+        echo "$portals"
+    done
+else
+    echo "iSCSI portals do not exist, so exiting"
+    exit 1
+fi
+
+echo "                         "
+
+for portals in "${PORTALS[@]}"
+do
+    IP=$(echo "$portals" | cut -f1 -d":")
+    if nmap -sS -p 3260 "${IP}" | grep 3260 | grep -q open
+    then
+        echo "$IP is open"
+    else
+        echo "$IP is not open"
+    fi
+done
+
+echo "*******************************"
+
+# Verify DNS SRV and A records exist for the worker respective of the iSCSI portals
+
+
+host=$(kubectl get vs -n sysmgmt-health cray-sysmgmt-health-grafana -o jsonpath='{.spec.hosts[0]}' | sed -e 's/grafana\.[^.]*\.//')
+
+dig -t SRV +short _sbps-hsn._tcp."${host}" _sbps-nmn._tcp."${host}" >> tmp_file
+
+if [ -s tmp_file ];then
+    echo "DNS SRV records exist as below:"
+    echo "---------------------------"
+    cat tmp_file
+    echo "*******************************"
+else
+    echo "DNS SRV records do not exist, please create them"
+    exit 1
+fi
+
+SRV=($(cat tmp_file | awk '{print $(NF)}' | sed -e 's/\.$//'))
+
+echo "DNS A records are as below"
+echo "--------------------------"
+
+for i in "${SRV[@]}"
+do
+    dig -t A +short "${i}"
+done
+
+rm tmp_file
+
+# Mapping between DNS A records and host iscsi portals
+echo "                                       "
+echo "*********************************************"
+echo "Mapping between DNS A records and host iscsi portals"
+echo "*********************************************"
+
+for s in $(dig -t srv +short _sbps-hsn._tcp."${host}" | sort -k3 | awk '{print $NF;}' | xargs); do printf "** $s **\n"; dig +short $s; done
+
+for s in $(dig -t srv +short _sbps-nmn._tcp."${host}" | sort -k3 | awk '{print $NF;}' | xargs); do printf "** $s **\n"; dig +short $s; done
```

Comment:
Are you looking for an error with these commands or are they purely informative?

Reply:
This gets the mapping between DNS A records and host iSCSI portals. Prior to this step DNS SRV and A records are retrived. So it won't reach here if these don't exist.